File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/JwtCookieConfigurationProperties.java
package uk.gov.justice.digital.hmpps.oauth2server.security;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import java.time.Duration;

@Configuration
@ConfigurationProperties(prefix = "jwt.cookie")
@Data
public class JwtCookieConfigurationProperties {
    private String name;
    private Duration expiryTime;
}

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/JwtCookieConfigurationProperties.kt
package uk.gov.justice.digital.hmpps.oauth2server.security

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import java.time.Duration

@ConstructorBinding
@ConfigurationProperties(prefix = "jwt.cookie")
data class JwtCookieConfigurationProperties(
  val name: String,
  val expiryTime: Duration,
)
